### PR TITLE
(ux-improvements) New: address book, fee, confirm intention, view of notifications

### PR DESCRIPTION
**Pull request by yazadzhy**

### CODE REVIEW

File: frontend/src/ui/wallet/transfer/destination-hints-view.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ export default observer(function DestinationHintsView({transfer}) {
     const [editorVisible, setEditorVisible] = useState(false)
     const [title, setTitle] = useState('')
     const destinationSuggestions = getDestinationSuggestions(destinationInputValue)
+    const destinationIsOwnAccount = toJS(accountManager.accounts).findIndex(account => account.publicKey.startsWith(destination)) !== -1
```
Comment (orbitlens) [inline frontend/src/ui/wallet/transfer/destination-hints-view.js:16]:
```suggestion
    const destinationIsOwnAccount = accountManager.accounts.some(account => account.publicKey.startsWith(destination))
```

File: frontend/src/ui/wallet/transfer/destination-hints-view.js [hunk 2]
```diff
@@ -30,18 +32,35 @@ export default observer(function DestinationHintsView({transfer}) {
             })
         //hide editor
         setEditorVisible(false)
-    }, [transfer])
+    }, [transfer, title])
```
Comment (orbitlens) [inline frontend/src/ui/wallet/transfer/destination-hints-view.js:35]: Not a good idea. The callback will be recreated each time title changes.

File: frontend/src/ui/wallet/transfer/transfer-tx-builder.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ export async function prepareTransferTx(transfer) {
 
     const builder = new TransactionBuilder(accountData, {
         networkPassphrase: resolveNetworkParams({network: transfer.network}).network,
-        fee: transfer.fee ? transfer.fee : await estimateFee(transfer.network)
+        fee: (['low', 'normal', 'high'].includes(transfer.fee)) ? await estimateFee(transfer.network, transfer.fee) : transfer.fee
```
Comment (orbitlens) [inline frontend/src/ui/wallet/transfer/transfer-tx-builder.js:18]: 👍

File: frontend/src/ui/wallet/transfer/transfer-view.js [hunk 4]
```diff
@@ -53,20 +52,19 @@ function TransferView() {
...
 
-    return <WalletOperationsWrapperView title="Transfer" action="Transfer" disabled={disabled}
-                                        transfer={transfer}
-                                        prepareTransaction={() => transfer.prepareTransaction()}
-                                        onFinalize={onFinalize}>
+    const prepareTransaction = useCallback(() => transfer.prepareTransaction(), [transfer])
+
+    const changeSlippage = useCallback(v => transfer.setSlippage(v), [transfer])
+
+    const cancelDestinationCreation = useCallback(() => runInAction(() => transfer.createDestination = false), [transfer])
```
Comment (orbitlens) [inline frontend/src/ui/wallet/transfer/transfer-view.js:65]:
```suggestion
    const cancelDestinationCreation = useCallback(() => runInAction(() => {
        transfer.createDestination = false
    }), [transfer])
```

File: frontend/src/util/account-notifications.js [hunk 5]
```diff
@@ -42,7 +42,7 @@ export function getNewPaymentsSince(network, account, ledger) {
                     return payments
                 iterations++
                 return fetchPayments(records[records.length - 1].paging_token)
-            })
+            }).catch(e => console.log(e))
```
Comment (orbitlens) [inline frontend/src/util/account-notifications.js:45]:
```suggestion
            }).catch(e => console.error(e))
```

File: frontend/src/util/fee-estimator.js
Comment (orbitlens) [inline frontend/src/util/fee-estimator.js:1]: 👍

File: frontend/src/util/get-federation-address.js
Comment (orbitlens) [inline frontend/src/util/get-federation-address.js:1]: This functionality duplicates `frontend/src/util/federation-address-resolver.js`

**Review by orbitlens — CHANGES_REQUESTED** [review]

File: frontend/src/ui/authentication/credentials-request-view.js [hunk 6]
```diff
@@ -19,55 +19,64 @@ export default function CredentialsRequestView({
...
+        if ((password || '').length < 8)
+            return 'Password too short'
+        if (requestPasswordConfirmation && password !== confirmation)
+            return 'Passwords do not match'
+        return null
+    }, [requestPasswordConfirmation])
 
-    function confirm() {
-        const validationError = validate()
+    const confirm = useCallback(() => {
```
[the remark this answers is not in the document]
Comment (yazadzhy) [reply frontend/src/ui/authentication/credentials-request-view.js:45]: `password` deleted from dependencies and all works like usually, but if delete `updateState` - doesn't work, seems cached empty password from default value, i'll check more deeply

File: frontend/src/ui/notifications/notifications.js [hunk 7]
```diff
@@ -66,17 +70,41 @@ function Notification({id, type = 'info', message = '', onDelete}) {
...
-    return <div className={cn('container-smoothly', {grow: !isClosing})}>
-        <div className={cn('segment notification', type, {slideIn: !isClosing}, {slideOut: isClosing})}>
-            <i className={cn('icon', iconDepends[type])}/>
+    return <div className={cn('notification-wrap', `notification-${type}`, {grow: !isClosing})}>
+        <div ref={notification} className={cn('notification', {slideIn: !isClosing}, {slideOut: isClosing})}>
+            <i className={cn('notification-icon', contextIcon[type])}/>
             <div className="text-small">{message}</div>
-            <div className="delete" onClick={close}><i className="icon icon-cancel"/></div>
+            <div className="delete" onClick={close}><i className="notification-icon icon-cancel"/></div>
+            <div className="lifetime" style={{animationDuration: timeToClose + 'ms'}}/>
```
[the remark this answers is not in the document]
Comment (yazadzhy) [reply frontend/src/ui/notifications/notifications.js:107]: Pausing of animation declared in CSS on :hover - it works with hover, selection of text, click on notification. I tested logic and visualization - all work correctly

File: frontend/src/ui/notifications/notifications.js [hunk 8]
```diff
@@ -66,17 +70,41 @@ function Notification({id, type = 'info', message = '', onDelete}) {
...
+        start.current = Date.now()
+        timer.current = window.setTimeout(close, remain.current)
+    }, [start, remain, close])
 
     useEffect(() => {
-        setTimeout(close, timeToClose)
-    }, [])
+        timer.current = setTimeout(close, timeToClose)
+
+        notification.current.addEventListener("mouseover", pauseTimer, true)
```
[the remark this answers is not in the document]
Comment (yazadzhy) [reply frontend/src/ui/notifications/notifications.js:93]: i tried it firstly, but not fully work like needed, but now all works correctly, strange, mb used wrong logic before

**Review by orbitlens — APPROVED** [review]

**Pull request merged**